zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java

```diff
@@ -0,0 +1,651 @@
+            // ZxidRange will be [low, high] where high will be the zxid right before the next
+            // snapshot's lastProcessedZxid
+            long nextSnapshotStartZxid =
+                Util.getZxidFromName(candidateSnapshots.get(i + 1).getName(), Util.SNAP_PREFIX);
+            zxidRange = new ZxidRange(zxidRange.getLow(), nextSnapshotStartZxid - 1);
+          }
+        }
+
+        filesToBackup.add(new BackupFile(f, false, zxidRange));
+      }
+    }
+
+    protected void endIteration(boolean errorFree) {
+      filesToBackup.clear();
+    }
+
+    protected BackupFile getNextFileToBackup() {
+      if (filesToBackup.isEmpty()) {
+        return null;
+      }
+
+      return filesToBackup.remove(0);
+    }
+
+    protected void backupComplete(BackupFile file) throws IOException {
+      synchronized (backupStatus) {
+        backupPoint.setSnapZxid(file.getMinZxid());
+        backupStatus.update(backupPoint);
+      }
+      backupStats.incrementNumberOfSnapshotFilesBackedUpThisIteration();
+
+      logger.info("Updated backedup snap zxid to {}",
+          ZxidUtils.zxidToString(backupPoint.getSnapZxid()));
+    }
+  }
+
+  /**
+   * Constructor for the BackupManager.
+   * @param snapDir the snapshot directory
+   * @param dataLogDir the txnlog directory
+   * @param serverId the id of the zk server
+   * @param backupConfig the backup config object
+   * @throws IOException
+   */
+  public BackupManager(File snapDir, File dataLogDir, long serverId, BackupConfig backupConfig)
+      throws IOException {
+    logger = LoggerFactory.getLogger(BackupManager.class);
+    logger.info("snapDir={}", snapDir.getPath());
+    logger.info("dataLogDir={}", dataLogDir.getPath());
+    logger.info("backupStatusDir={}", backupConfig.getStatusDir().getPath());
+    logger.info("tmpDir={}", backupConfig.getTmpDir().getPath());
+    logger.info("backupIntervalInMinutes={}", backupConfig.getBackupIntervalInMinutes());
+    logger.info("serverId={}", serverId);
+    logger.info("namespace={}", backupConfig.getNamespace());
+
+    this.snapDir = snapDir;
+    this.dataLogDir = dataLogDir;
+    this.backupConfig = backupConfig;
+    this.tmpDir = backupConfig.getTmpDir();
+    this.backupStatus = new BackupStatus(backupConfig.getStatusDir());
+    this.backupIntervalInMilliseconds =
+        TimeUnit.MINUTES.toMillis(backupConfig.getBackupIntervalInMinutes());
+    this.serverId = serverId;
+    this.namespace = backupConfig.getNamespace() == null ? "UNKNOWN" : backupConfig.getNamespace();
+    try {
+      backupStorage = BackupUtil.createStorageProviderImpl(backupConfig);
+    } catch (ReflectiveOperationException e) {
+      throw new BackupException(e.getMessage(), e);
+    }
+    initialize();
+  }
+
+  /**
+   * Start the backup processes.
+   * @throws IOException
+   */
+  public synchronized void start() throws IOException {
+    logger.info("BackupManager starting.");
+
+    (new Thread(logBackup)).start();
```

Comment:
Naming these threads will help in debugging.
Also, explicitly set the daemon flag based on the type of the thread.